src/docfx/template/TemplateEngine.cs

```diff
@@ -63,6 +66,19 @@ public RenderType GetRenderType(ContentType contentType, SourceInfo<string?> mim
             };
         }
 
+        public string? GetThemeBranch()
+        {
+            if (_package is LocalPackage package)
+            {
+                var (themeRepo, _) = _repositoryProvider.GetRepository(PathString.DangerousCreate($"{package.PackagePath}/"));
```

Comment:
> _repositoryProvider.GetRepository(PathString.DangerousCreate($"{package.PackagePath}/")); [](start = 36, length = 90)

This looks like internal logic of `LocalPackage`. Would it be better let package returns branch info?

Reply:
but the package is not always a repository, even for localPackage, it is more related to the template, so I put it here.